forch/faucet_event_client.py

```diff
@@ -105,23 +105,26 @@ def has_event(self, blocking=False):
             else:
                 return False
 
-    def _filter_faucet_event(self, event):
+    def _filter_faucet_event(self, event, target_event):
```

Comment:
this method needs some work since it's either misnamed or doing too many different things.  It's now taking two events, which really muddles the picture. Also, the parameters are confusing.  I would take a look at it and maybe break it out into the necessary component parts.  E.g. one that's managing the event sequence checking, another for the debounce logic.

Reply:
could the new functions still be called from within the filter_faucet_event method. It uses it as a check for whether to dispatch the event or not.

Reply:
Or I could check each of the new methods. Fair.

Reply:
Leaving in event order and target field check.

Reply:
I think you can simplify by removing target_event here, and doing the check where it's called... -- can it just be

dispatch = self._check_event_order(event) and target_event

?  Seems like this method is unnecessary.